tests/pravega_reader_test.py

```diff
@@ -320,3 +321,70 @@ async def test_largeEvents(self):
                 count+=1
                 self.assertEqual(b'a'*100000, event.data(), "Invalid event data")
             r1.release_segment(segment_slice)
+
+    # This test is to verify get_streamcut API.
+    # It creates a Pravega stream and writes 4 events.
+    # Add reader to ReaderGroup which read only first 2 events and release segment slice.
+    # Reader positions are updated for every 3 secs on acquistion of segments or release segment call so pause for 3 sec before release segment call.
+    # Get streamcut which has start offset points to 3rd event. Use this streamcut to create new RG
+    # Now reader in this RG reads remaining to 2 events.
+    async def test_getStreamcut(self):
+        suffix = str(random.randint(0, 100))
+        scope = "testReaderSC"
+        stream = "testStream" + suffix
+        print("Creating a Stream Manager, ensure Pravega is running")
+        stream_manager = pravega_client.StreamManager("tcp://127.0.0.1:9090")
+        pravega_client.StreamReaderGroupConfig(True, scope, stream)
+        print("Creating a scope")
+        scope_result = stream_manager.create_scope(scope)
+        print(scope_result)
+        print("Creating a stream ", stream)
+        stream_result = stream_manager.create_stream(scope, stream, 1)
+        print(stream_result)
+        w1 = stream_manager.create_writer(scope, stream)
+
+        print("Write 4 events")
+        w1.write_event("test event1")
+        w1.write_event("test event2")
+        w1.write_event("test event3")
+        w1.write_event("test event4")
+        w1.flush()
+
+        # Create a reader Group Configuration to read from HEAD of stream.
+        rg_config = pravega_client.StreamReaderGroupConfig(False, scope, stream)
+        reader_group=stream_manager.create_reader_group_with_config("rg" + suffix, scope, rg_config)
+
+        r1 = reader_group.create_reader("reader-1")
+        segment_slice = await r1.get_segment_slice_async()
+        print(segment_slice)
+
+        # consume the segment slice for events. Read single event ie event0 and get ReaderStreamcut.
+        # Same should be used to construct readerGroup2 and that should start reading from event 2 ie event1
+        count=0
+        for event in segment_slice:
+            count+=1
+            print(event.data())
+            print("event offset is ")
+            print(event.offset())
+            if count == 2 :
+                break
+        self.assertEqual(count, 2, "Two events are expected")
+        time.sleep(3)
```

Comment:
We should not depend on time, especially not in a hard coded way like this. Maybe we could have a `#[cfg(test)]` method which forces the update